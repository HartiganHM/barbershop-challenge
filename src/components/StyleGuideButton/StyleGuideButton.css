@import url("https://fonts.googleapis.com/css?family=Open+Sans");
@import url("https://fonts.googleapis.com/css?family=Open+Sans");
.StyleGuideButton {
  display: flex;
  align-items: center;
  justify-content: space-around;
  height: 4rem;
  padding: 0.5rem;
  border-radius: 2px;
  background-color: #fff;
  color: #2F80ED;
  font-size: 1.6rem;
  transition: 0.2s ease-in-out;
  cursor: pointer;
  background-color: #2F80ED;
  color: #fff;
  cursor: default;
  align-self: center;
  display: flex;
  justify-content: center;
  width: 18rem;
  margin: 2rem;
  cursor: pointer;
  text-decoration: none; }
  .StyleGuideButton .style-icon {
    width: 16px;
    margin: 0.5rem;
    margin-right: 1.5rem;
    fill: rgba(255, 255, 255, 0.7);
    transition: 0.2s; }
  .StyleGuideButton:hover {
    background-color: #33ADFF;
    color: #fff; }
    .StyleGuideButton:hover .style-icon {
      fill: #fff;
      transform: rotate(-90deg); }
  .StyleGuideButton:active {
    background-color: #3F67D4;
    color: #fff; }
