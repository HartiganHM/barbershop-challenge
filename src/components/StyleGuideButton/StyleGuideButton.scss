@import '../../styles/variables.scss';
@import '../../styles/mixins.scss';
.StyleGuideButton {
  @include buttonDefaultAlternate;
  width: 18rem;
  margin: 2rem;
  cursor: pointer;
  .style-wrapper {
    display: flex;
    height: 3rem;
    align-items: center;
    text-decoration: none;
    color: #fff;
    .style-icon {
      width: 16px;
      margin: 0.5rem;
      fill: $white-transparent;
      transition: 0.2s;
    }
    .style-text {
      margin-left: 1rem;
    }
  }
  &:hover {
    @include buttonHover;
    .style-icon {
      fill: #fff;
      transform: rotate(-90deg);
    }
  }
  &:active {
    @include buttonActive;
  }
}
