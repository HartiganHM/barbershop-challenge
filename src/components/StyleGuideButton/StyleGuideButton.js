import React from 'react';
import { Link } from 'react-router-dom';
import iconData from '../../data/iconData';
import './StyleGuideButton.css';

const StyleGuideButton = () => {
  return (
    <Link to="/styleguide">
      <svg viewBox="0 0 19.3 19.3" className="style-icon">
        <path d={iconData.style} />
      </svg>
      <span className="home-text">Style</span>
    </Link>
  );
};

export default StyleGuideButton;
