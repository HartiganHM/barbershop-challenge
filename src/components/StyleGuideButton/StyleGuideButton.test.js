import React from 'react';
import { shallow } from 'enzyme';
import StyleGuideButton from './StyleGuideButton';

describe('StyleGuideButton tests', () => {
  it('Should match the snapshot', () => {
    const renderStyleGuideButton = shallow(<StyleGuideButton />);

    expect(renderStyleGuideButton).toMatchSnapshot();
  });
});
