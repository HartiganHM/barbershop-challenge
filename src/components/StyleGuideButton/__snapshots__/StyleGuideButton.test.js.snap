// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyleGuideButton tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StyleGuideButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        className="style-wrapper"
        replace={false}
        to="/styleguide"
      >
        <svg
          className="style-icon"
          viewBox="0 0 19.3 19.3"
        >
          <path
            d="M7.6,10.3c-0.4,0.4-0.5,0.7-0.9,1.5C7,11.9,7.6,12.4,8,13.3c0.8-0.5,1.2-0.5,1.7-0.9c2.9-2.4,9.9-11,9.6-11.3
    C19,0.8,10.1,7.5,7.6,10.3z M5.9,12.4c-1.2-0.2-2.5,0.6-3.3,2.5c-0.8,2-2.3,2.8-2.6,2.7c1.5,0.6,6.2,2,7.3-3.4
    C6.9,12.9,5.9,12.4,5.9,12.4z"
          />
        </svg>
        <input
          className="style-text"
          type="submit"
          value="Style Guide"
        />
      </Link>,
      "className": "StyleGuideButton",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <svg
            className="style-icon"
            viewBox="0 0 19.3 19.3"
          >
            <path
              d="M7.6,10.3c-0.4,0.4-0.5,0.7-0.9,1.5C7,11.9,7.6,12.4,8,13.3c0.8-0.5,1.2-0.5,1.7-0.9c2.9-2.4,9.9-11,9.6-11.3
    C19,0.8,10.1,7.5,7.6,10.3z M5.9,12.4c-1.2-0.2-2.5,0.6-3.3,2.5c-0.8,2-2.3,2.8-2.6,2.7c1.5,0.6,6.2,2,7.3-3.4
    C6.9,12.9,5.9,12.4,5.9,12.4z"
            />
          </svg>,
          <input
            className="style-text"
            type="submit"
            value="Style Guide"
          />,
        ],
        "className": "style-wrapper",
        "replace": false,
        "to": "/styleguide",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <path
              d="M7.6,10.3c-0.4,0.4-0.5,0.7-0.9,1.5C7,11.9,7.6,12.4,8,13.3c0.8-0.5,1.2-0.5,1.7-0.9c2.9-2.4,9.9-11,9.6-11.3
    C19,0.8,10.1,7.5,7.6,10.3z M5.9,12.4c-1.2-0.2-2.5,0.6-3.3,2.5c-0.8,2-2.3,2.8-2.6,2.7c1.5,0.6,6.2,2,7.3-3.4
    C6.9,12.9,5.9,12.4,5.9,12.4z"
            />,
            "className": "style-icon",
            "viewBox": "0 0 19.3 19.3",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "d": "M7.6,10.3c-0.4,0.4-0.5,0.7-0.9,1.5C7,11.9,7.6,12.4,8,13.3c0.8-0.5,1.2-0.5,1.7-0.9c2.9-2.4,9.9-11,9.6-11.3
    C19,0.8,10.1,7.5,7.6,10.3z M5.9,12.4c-1.2-0.2-2.5,0.6-3.3,2.5c-0.8,2-2.3,2.8-2.6,2.7c1.5,0.6,6.2,2,7.3-3.4
    C6.9,12.9,5.9,12.4,5.9,12.4z",
            },
            "ref": null,
            "rendered": null,
            "type": "path",
          },
          "type": "svg",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "style-text",
            "type": "submit",
            "value": "Style Guide",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": [Function],
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          className="style-wrapper"
          replace={false}
          to="/styleguide"
        >
          <svg
            className="style-icon"
            viewBox="0 0 19.3 19.3"
          >
            <path
              d="M7.6,10.3c-0.4,0.4-0.5,0.7-0.9,1.5C7,11.9,7.6,12.4,8,13.3c0.8-0.5,1.2-0.5,1.7-0.9c2.9-2.4,9.9-11,9.6-11.3
    C19,0.8,10.1,7.5,7.6,10.3z M5.9,12.4c-1.2-0.2-2.5,0.6-3.3,2.5c-0.8,2-2.3,2.8-2.6,2.7c1.5,0.6,6.2,2,7.3-3.4
    C6.9,12.9,5.9,12.4,5.9,12.4z"
            />
          </svg>
          <input
            className="style-text"
            type="submit"
            value="Style Guide"
          />
        </Link>,
        "className": "StyleGuideButton",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <svg
              className="style-icon"
              viewBox="0 0 19.3 19.3"
            >
              <path
                d="M7.6,10.3c-0.4,0.4-0.5,0.7-0.9,1.5C7,11.9,7.6,12.4,8,13.3c0.8-0.5,1.2-0.5,1.7-0.9c2.9-2.4,9.9-11,9.6-11.3
    C19,0.8,10.1,7.5,7.6,10.3z M5.9,12.4c-1.2-0.2-2.5,0.6-3.3,2.5c-0.8,2-2.3,2.8-2.6,2.7c1.5,0.6,6.2,2,7.3-3.4
    C6.9,12.9,5.9,12.4,5.9,12.4z"
              />
            </svg>,
            <input
              className="style-text"
              type="submit"
              value="Style Guide"
            />,
          ],
          "className": "style-wrapper",
          "replace": false,
          "to": "/styleguide",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <path
                d="M7.6,10.3c-0.4,0.4-0.5,0.7-0.9,1.5C7,11.9,7.6,12.4,8,13.3c0.8-0.5,1.2-0.5,1.7-0.9c2.9-2.4,9.9-11,9.6-11.3
    C19,0.8,10.1,7.5,7.6,10.3z M5.9,12.4c-1.2-0.2-2.5,0.6-3.3,2.5c-0.8,2-2.3,2.8-2.6,2.7c1.5,0.6,6.2,2,7.3-3.4
    C6.9,12.9,5.9,12.4,5.9,12.4z"
              />,
              "className": "style-icon",
              "viewBox": "0 0 19.3 19.3",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "d": "M7.6,10.3c-0.4,0.4-0.5,0.7-0.9,1.5C7,11.9,7.6,12.4,8,13.3c0.8-0.5,1.2-0.5,1.7-0.9c2.9-2.4,9.9-11,9.6-11.3
    C19,0.8,10.1,7.5,7.6,10.3z M5.9,12.4c-1.2-0.2-2.5,0.6-3.3,2.5c-0.8,2-2.3,2.8-2.6,2.7c1.5,0.6,6.2,2,7.3-3.4
    C6.9,12.9,5.9,12.4,5.9,12.4z",
              },
              "ref": null,
              "rendered": null,
              "type": "path",
            },
            "type": "svg",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "style-text",
              "type": "submit",
              "value": "Style Guide",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": [Function],
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
